### ISSUE TITLE
Trade confirmation doesn't work with amounts over 255(applicable to Gil trades)

### ISSUE THREAD

**Issue by Hokuten85** [DarkstarProject/darkstar#6282]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 
0.0.8319

**_Source Branch_** (master/stable) **:** 
master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
Should be able to reproduce this by attempting to confirmSlot or confirmItem on a trade involving Gil with an amount over 255, and then call trade:confirmTrade(). All of the amount parameters in the functions involved are defined as uint8.

src/map/trade_container.cpp
getConfirmedStatus
setConfirmedStatus

src/map/trade_container.h
getConfirmedStatus
setConfirmedStatus
std::vector<uint32>	    m_confirmed;

**Comment by SirGouki** [comment]
The reason is the data type UINT8, range is between 0 and 255.  Type needs to be changed and proper value checking should be implemented.  I would do it, but I'm trying to update everything right now, and I have to set things back up for a client for darkstar.

**Comment by Hokuten85** [comment]
Yep. I've corrected this on my own server, but due to all the other funky customizations I've made it makes pull requests difficult.

**Comment by KnowOne134** [comment]
Clone dsp again, make a new branch and pull request. Or change branches on your own server to newest dsp

**Comment by TeoTwawki** [comment]
in getConfirmedStatus that's the slot ID not the amount, so that one should be uint8 still.

**Comment by TeoTwawki** [comment]
https://github.com/DarkstarProject/darkstar/pull/6382/files?utf8=%E2%9C%93&diff=unified&w=1
